docs/CalicoSwarm.md

```diff
@@ -0,0 +1,152 @@
+  - 2 Swarm nodes (to host our containers)
+
+## Installing Swarm on your cluster
+Our Docker Swarm will consist of 2 nodes and a Swarm manager.  We will use the fourth server as the client to control our Swarm.  We'll use the token-based discovery backend to configure our Swarm.
+
+First, let us create the token which will identify our swarm.  To do this, run the following on your client node:
+```
+docker pull swarm
+docker run --rm swarm create
+``` 
+
+The second command will print a unique cluster ID token which we must use to configure our Swarm.  We'll reference this value in the tutorial as ```<swarm_token>```. 
+Your token should look something like this: d435dc114a3dd89af2fae239257338ce
+
+Now that we've got a token, we can begin to configure our Swarm.  First, let's join each of our two Swarm nodes to the
+cluster.  Run the following commands on each node, replacing ```<swarm_token>``` with the token from above, and ```<node_ip>``` with the IP address of this node. 
+```
+docker run -d swarm join --addr=<node_ip>:2377 token://<swarm_token>
+```
+
+Let's now configure the Swarm manager node.  To do this, run the following on your Swarm manager node.  Note that ```<swarm_port>``` in the following command can be any unused TCP port on the manager server.  This is the port the client will use to communicate with the Swarm manager daemon.
+```
+docker run -d -p <swarm_port>:2375 swarm manage token://<swarm_token>
+```
+
+At this point, the Swarm agents are running on our cluster.  However, the cluster won't be complete until Calico is installed and running on the Swarm nodes.
+
+## Installing etcd
+Calico requires etcd, so let's install it on our cluster.  For this example, we'll only configure a single node etcd cluster.  However, in a production environment, a minimum of a three node etcd cluster is recommended.
+
+Let's install etcd on our manager node.  Run the following to download the etcd binaries.
+```
+# Download etcd for Linux 
+curl -L https://github.com/coreos/etcd/releases/download/v2.0.11/etcd-v2.0.11-linux-amd64.tar.gz -o etcd-v2.0.11-linux-amd64.tar.gz
+tar xzvf etcd-v2.0.11-linux-amd64.tar.gz
+
+# Move the binaries to /usr/local/bin for easy access.
+cd etcd-v2.0.11-linux-amd64
+sudo cp etcd /usr/local/bin
+sudo cp etcdctl /usr/local/bin
+```
+
+Now that etcd is installed, let's run our single node cluster. We need to start etcd with the IP address of the manager
+node, so store it in an environment variable for easy access.
+```
+export MANAGER_IP=<Manager's IP address>
+```
+
+Use the following command to start etcd:
+```
+etcd -name etcd0  -advertise-client-urls http://$MANAGER_IP:2379,http://$MANAGER_IP:4001  -listen-client-urls http://0.0.0.0:2379,http://0.0.0.0:4001  -initial-advertise-peer-urls http://$MANAGER_IP:2380  -listen-peer-urls http://0.0.0.0:2380  -initial-cluster-token etcd-cluster-1  -initial-cluster etcd0=http://$MANAGER_IP:2380 -initial-cluster-state new &
+
+```
```

Comment:
Worth following up with a simple ps to verify etcd is running?  or some other way?